### DISCUSSION TITLE
Script to open URL in default browser?

### DISCUSSION THREAD

**Discussion by jsjoeio** [johnlindquist/kit#142] [Q&A] [answered]
After some googling, I am not having any luck finding an easy way to open URLs in the default browser. It seems like the only option is to:
1. get the [user's default browser](https://apple.stackexchange.com/questions/313454/applescript-find-the-users-set-default-browser)
2. use that application to open a link

Wondering if anyone knows of a better/easier solution?

**Comment by johnlindquist** [comment, marked as answer]
The bash "open" command uses default apps based on protocols of the system:

```js
let url = "https://jsjoe.io/";
exec(`open ${url}`)
```

**Reply by jsjoeio** [reply]
Awesome!!! Thanks @johnlindquist 🎉

Works with [Browserosaurus](https://wstone.io/browserosaurus) too!

https://user-images.githubusercontent.com/3806031/114595125-e3e2d380-9c42-11eb-8458-93b26f8f73b7.mov

https://user-images.githubusercontent.com/3806031/114595513-4fc53c00-9c43-11eb-8ea5-69ea1ec4d2af.mov

Might have to configure with [Finicky](https://github.com/johnste/finicky) too so I can make it use the default instead of prompt.

**Reply by johnlindquist** [reply]
Open specific app (and pass args if you want) looks like:

```bash
exec(`open -a /Applications/Firefox.app "https://jsjoe.io" --args --kiosk`)
```
